src/firefox_client.ts

```diff
@@ -0,0 +1,658 @@
+      // As you can tell, many of these will take a too much many and the systme will fail. This is what we do here
+      // This means that it will also close the firefox client a user may be using
+      const p = Deno.run({
+        cmd: ["taskkill", "/F", "/IM", "firefox.exe"],
+        stdout: "null",
+        stderr: "null",
+      });
+      await p.status();
+      p.close();
+    }
+    if (errMsg) {
+      throw new Error(errMsg);
+    }
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // FILE MARKER - METHODS - PRIVATE ///////////////////////////////////////////
+  //////////////////////////////////////////////////////////////////////////////
+
+  /**
+   * Our async iterator to handle reading packets from the connection,
+   * and parsing them into a usable format
+   *
+   * @returns An async iterator to get the next packet from
+   */
+  private async readPackets(): Promise<Packet> {
+    let partial = "";
+    let packet;
+    for await (const chunk of Deno.iter(this.conn)) { // NOTE: All packets seems to have an id associated to them, for their type, so evaluateJSAsync will always return `74:{...}` and the evaluation result will always have an id of 321
+      const decodedChunk = decoder.decode(chunk);
+      const rawPackets = decodedChunk
+        .split(/[0-9]{1,4}:{/) // split and get rid of the ids so each item should be parsable json
+        .filter((packet) => packet !== "")
+        .map((msg, i) => {
+          // We want to re add the {, but in the instance `message` is `d":4}125:{...}`, we should add it to the first packet because its the last aprt of a partial, so if we do, then we can't combine them together
+          if (i === 0) {
+            // We'll try parse it, if we cant then we know its a partial, so dont add the bracket
+            try {
+              JSON.parse("{" + msg);
+              return `{${msg}`;
+            } catch (err) {
+              return msg;
+            }
+          } else {
+            return "{" + msg;
+          }
+        }); // add back the `{` that we removed above, so we can still easily parse it
+      // Turn the packets into json, if it fails then it means a packet is partial, so we save it
+      const packets = rawPackets.map((packet) => {
+        try {
+          return JSON.parse(packet);
+        } catch (err) {
+          partial += packet;
+        }
+      }).filter((packet) => packet !== undefined);
+      // Then try check if partial is full, if it is then add it to the packets (at start, because remember this packet still originally came first
+      try {
+        const j = JSON.parse(partial);
+        partial = "";
+        packets.unshift(j);
+      } catch (err) {
+        // still a partial, do nothing
+      }
+
+      const validPackets = packets.filter((packet) => {
+        if (UNSOLICITED_EVENTS.includes(packet.type) === true) {
```

Comment:
so `UNSOLICITED_EVENTS` are silently ignored? maybe we can provide an FF doc link on line 3 so the events are easily accessible

Reply:
If only I knew what the events were xD that unsolicited events variable was actually pulled from a node Firefox package, and I just added to it when I found an event we don’t care about, but I’ll try dig around for some docs

Reply:
hmm... since it came from a node package, maybe we just state what node package it came from and that they're ignored

Reply:
Yeah alright sounds fair

Reply:
I can't even find their source code for it now, maybe they removed it but im going to add a note anyways

Reply:
okie doke

Reply:
https://github.com/drashland/sinco/pull/52/commits/020ae1577d5f44242069c780adfe4a2a303ce999